### ISSUE TITLE
Tasks and Che Commands should not use "machineName"

### ISSUE THREAD

**Issue by slemeur** [eclipse-che/che#13667]
### Description
"Machine" has been deprecated and should not be used anywhere in Che anymore.

````
{
            "label": "run the web app 2",
            "source": "che",
            "type": "che",
            "command": "cd ${CHE_PROJECTS_ROOT}/nodejs-web-app/app && nodemon app.js",
            "target": {
                "machineName": "nodejs"
            }
}
````

<img width="1758" alt="Capture d’écran 2019-07-02 à 11 51 20" src="https://user-images.githubusercontent.com/1636769/60504663-4c58fe80-9cc2-11e9-8a1f-228996b30a04.png">

We still have it for tasks, so it has to be adapted before the Che 7 GA in order to introduce a formalism which will need to be backward compatible after

**Comment by RomanNikitenko** [comment]
@slemeur 
Could you provide the new field name we should use instead of `machineName`?

If I understand correctly at least for 3 places changes are required: 
- task plugin
- machine exec 
- code which serves devfile 

thank you!

**Comment by slemeur** [comment]
Could it just be made consistent with the formalism used in devfile?

**Comment by skabashnyuk** [comment]
>Could it just be made consistent with the formalism used in devfile? 

That’s hard. Since devfile are not representing running workspace. In this case, machineName is the name of the running container. For workspace runtime, we still use Che6 like objects and concepts.

**Comment by l0rd** [comment]
@skabashnyuk in devfile we use the components alias for that purpose. Not sure why it can't apply it here. And workspace runtime using Che6 objects and concepts should not be taken as a example for new, che 7 only, objects (i.e. che-theia tasks).

**Comment by RomanNikitenko** [comment]
At the moment task plugin gets info about target container as attribute of command `machineName` and provides this one for machine exec.
So I guess we can: 
- display for user that field as `component` or `containerName` for example, but in code still use 'machineName' - only for task-plugin changes are required
- clean up code and use the new field for user and in our code as well - for task-plugin, machine exec and for che objects changes are required

**Comment by l0rd** [comment]
@RomanNikitenko I think we should do that in 2 steps. First step for 7.0.0 and second step for 7.1.0

**Comment by RomanNikitenko** [comment]
OK for me to do in 2 steps.

But what field we should use instead of `machineName`? 
`component`? `containerName`? 
@l0rd @slemeur @skabashnyuk

**Comment by slemeur** [comment]
Let's be consistent with the devfile formalism please.
https://redhat-developer.github.io/devfile/devfile#components

**Comment by skabashnyuk** [comment]
@slemeur as I remember yesterday on a meeting with you and @l0rd we decided to use "containerName" . The motivation was a workspace panel that is referencing to containers.

**Comment by RomanNikitenko** [comment]
I provided the PR and I use the `containerName` instead of `machineName` field for che tasks.

**Comment by RomanNikitenko** [comment]
The PR for first step was merged to 7.0.0 branch.
For second step I created [another issue](https://github.com/eclipse/che/issues/14081), which affect at least task-plugin, machine exec and che objects.
